### PR TITLE
Nobs-declining-payment

### PR DESCRIPTION
**Pull request by darmhoo**
#### What does this PR do?
- (replace me) Explain WHY this PR is required. E.g.: What feature it adds, what problem it solves, …

#### Description of Task proposed in this pull request?
- (replace me) A description of WHAT changed in the codebase. Additionally add the reasoning if it's complex or abstract. 

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [Story ID:  replace text with story id](paste Pivotal Tracker link here)

#### Any background context you want to add (Operations Impact)?
- (replace me) List the steps that must be taken for this PR to work.
E.g.: `npm install`, `Add "github_key" to environment variables, ...`

Be sure to include all systems that needs to be changed or which system is
affected by the change (Ex: Requires Styled Components to be installed and
configured in `babel.rc`, file to be updated).

#### What I have learned working on this feature: [If you don't put anything here, you are doing it wrong!]
- (replace me) I learned how to create a pull request template

### CODE REVIEW

File: app/Amortization/Amortization.php [hunk 1]
```diff
@@ -83,17 +83,43 @@ public function create()
     public function preview()
     {
         $IsSuperLoan = Str::contains($this->order->businessType->slug, 'super');
+        $IsNoBs = Str::contains($this->order->businessType->slug, 'no_bs');
+        $IsNoBsRenewalLoan = Str::contains($this->order->businessType->slug, 'renewal');
```
Comment (teebarg) [inline app/Amortization/Amortization.php:87]: Don't we have other business types that includes `renewal` that isn't NoBs?

File: app/Amortization/Amortization.php [hunk 2]
```diff
@@ -83,17 +83,43 @@ public function create()
...
+
+
         if ($IsSuperLoan && env('USE_SUPER_LOAN_CALC')) {
             return $this->getSuperLoaPaymentPlans();
+        } else if ($IsNoBs) {
```
Comment (teebarg) [inline app/Amortization/Amortization.php:92]: let's add an env just like superloan to turn if ON or OFF

File: app/Amortization/Amortization.php [hunk 3]
```diff
@@ -83,17 +83,43 @@ public function create()
...
+            if ($IsNoBsRenewalLoan) {
```
Comment (teebarg) [inline app/Amortization/Amortization.php:93]: What are we doing here?

File: app/Amortization/Amortization.php [hunk 4]
```diff
@@ -110,6 +136,39 @@ private function getSuperLoaPaymentPlans()
         }
         return $plan;
     }
+    private function getNoBsNewPaymentPlans()
```
Comment (teebarg) [inline app/Amortization/Amortization.php:139]: `getNoBsNewPaymentPlans` and `getNoBsRenewalPaymentPlans` looks similar, can't we reuse logic?

File: app/Amortization/Amortization.php (hunk 3 above)
Comment (darmhoo) [reply app/Amortization/Amortization.php:93]: checks if it is nobs first, then check if renewal or not

File: app/Amortization/Amortization.php [hunk 5]
```diff
@@ -84,23 +84,17 @@ public function preview()
     {
         $IsSuperLoan = Str::contains($this->order->businessType->slug, 'super');
         $IsNoBs = Str::contains($this->order->businessType->slug, 'no_bs');
```
Comment (harmlessprince) [inline app/Amortization/Amortization.php:86]: I believe there is slug for  no bs renewal check the db

Comment (darmhoo) [reply app/Amortization/Amortization.php:86]: ap_no_bs_renewal_non_verve, ap_no_bs_product_renewal_non_verve.
this is the challenge. i need to include the product too

Comment (darmhoo) [reply app/Amortization/Amortization.php:86]: that is y i am doing that

Comment (harmlessprince) [reply app/Amortization/Amortization.php:86]: okay now

**Review by teebarg — APPROVED** [review]

**Pull request merged**